### PR TITLE
Allure cucumber jvm

### PR DESCRIPTION
**Pull request by clicman**
Add allure-cucumber-jvm adaptor

### CODE REVIEW

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java [hunk 1]
```diff
@@ -0,0 +1,461 @@
+package io.qameta.allure.cucumberjvm;
+
+import cucumber.runtime.CucumberException;
+import cucumber.runtime.StepDefinitionMatch;
+import gherkin.I18n;
+import gherkin.formatter.Formatter;
+import gherkin.formatter.Reporter;
+import gherkin.formatter.model.*;
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java]: do not use wildcard imports

File: settings.gradle [hunk 2]
```diff
@@ -4,4 +4,4 @@ include 'allure-junit4'
 //TODO wait for junit5-M4
 //include 'allure-junit5'
 include 'allure-testng'
-
+include 'allure-cucumber-jvm'
```
Comment (baev) [inline settings.gradle]: add a new line in end of file

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java (hunk 1 above)
Comment (clicman) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java]: Damn Idea!

File: settings.gradle (hunk 2 above)
Comment (clicman) [inline settings.gradle]: Got it!

**Comment by baev** [comment]
you also need to fix all the pmd/checkstyle/findbugs issues

**Comment by clicman** [comment]
> you also need to fix all the pmd/checkstyle/findbugs issues
Is these check done automatically? I can fix it in right way only if it will hardly be declared rules or automatically checked by right configured sonar of smth. similar.

**Comment by baev** [comment]
simply run `./gradlew clean build` and you will see the errors in console

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+import gherkin.formatter.model.Tag;
+import io.qameta.allure.Allure;
+import io.qameta.allure.AllureLifecycle;
+
+import io.qameta.allure.model.Status;
+import io.qameta.allure.model.StatusDetails;
+import io.qameta.allure.model.StepResult;
+import io.qameta.allure.model.TestResult;
+
+import java.util.*;
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java]: imports

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/StepUtils.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
+package io.qameta.allure.cucumberjvm;
+
+import cucumber.runtime.CucumberException;
+import cucumber.runtime.StepDefinitionMatch;
+import gherkin.formatter.model.*;
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/StepUtils.java]: imports

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/Utils.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    private static MessageDigest getMessageDigest() {
+        try {
+            return MessageDigest.getInstance(MD_5);
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalStateException("Could not find md5 hashing algorithm", e);
+        }
+    }
+
+    public static String getStackTraceAsString(final Throwable throwable) {
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/Utils.java]: you can use `ResultsUtils.getStatusDetails()` instead

**Comment by baev** [comment]
@clicman thanks!

**Pull request merged**